ENH, TST: Add FutureWarning to read_table

kathleenhang opened the pull request:
- [x] closes #51017

kathleenhang committed:
Add FutureWarning to read_table

MarcoGorelli commented on pandas/tests/io/parser/test_parse_dates.py:
```diff
@@ -1290,17 +1290,17 @@ def test_parse_dates_empty_string(all_parsers):
 
 
+@pytest.mark.parametrize(
+    'reader',
```
You'll need to pass `reader` as an argument to the function, check the pytest docs on parametrisation

kathleenhang replied:
> Thanks for working on this!
> 
> I've just left some comments - did you manage to run the test locally?

I didn't think to do that since I am new to testing, but I have now. 

Thanks for your help!

MarcoGorelli commented on pandas/tests/io/parser/test_parse_dates.py:
```diff
@@ -1290,17 +1290,17 @@ def test_parse_dates_empty_string(all_parsers):
-    # GH 49024
+    # GH 49024, 51017
     parser = all_parsers
-    data = "Date,test\n2012-01-01,1\n,2"
```
This shouldn't go

kathleenhang replied:
If I parametrise over parser.read_csv_check_warnings and parser.read_table_check_warnings, how will I be able to insert data?

From what I'm understanding, getattr(parser, reader) should extract the attribute values of 'read_csv_check_warnings' or 'read_table_check_warnings' from the parser object depending on what is defined under our parametrization.

I'm having trouble understanding how the interchangeability works and how I would perform the data insertion if the value of reader is interchanging.

MarcoGorelli replied:
I was thinking something like
```
    data = "Date,test\n2012-01-01,1\n,2"
    getattr(parser, reader)(
        UserWarning,
        "The argument 'infer_datetime_format' is deprecated",
        StringIO(data),
        parse_dates=["Date"],
        infer_datetime_format=True,
    )
```
so that the first time you get `parser.read_csv_check_warnings`, and the second time `parser.read_table_check_warnings` - would that work?

MarcoGorelli replied:
ah, thanks - looks like we can't use this `getattr` trick then, else the stacklevel count gets messed up

I think we may just need to copy-and-paste the test below, so one test has `parser.read_csv_check_warnings` and the other one has `parser.read_table_check_warnings`

kathleenhang replied:
Actually, your original idea of parametrization does work. 

Pytest only fails when I am running it through my Docker virtual environment from my terminal app. However, it succeeds when I run it through the VSCode terminal.  I am running a Docker virtual environment through VSCode using the Dev Containers extension. 

I wonder how the two virtual environments are out of sync?

MarcoGorelli reviewed (changes requested):
Thanks for working on this! 

I've just left some comments - did you manage to run the test locally?

kathleenhang committed:
Update test function

kathleenhang committed:
Add warning type

MarcoGorelli commented on pandas/tests/io/parser/test_parse_dates.py:
```diff
@@ -1290,16 +1290,27 @@ def test_parse_dates_empty_string(all_parsers):
     data = "Date,test\n2012-01-01,1\n,2"
-    parser.read_csv_check_warnings(
-        UserWarning,
+    warning_type = (FutureWarning if reader == "read_table_check_warnings"
+                    else UserWarning)
```
these should both be `FutureWarning` - looks like the read_csv one was wrong to begin with, could you change that to be `FutureWarning` too please?

MarcoGorelli reviewed (changes requested):
almost there 💪

MarcoGorelli commented:
you'll also need to run the code checks, see here for how to run them https://pandas.pydata.org/docs/dev/development/contributing_codebase.html#pre-commit (and of course do reach out if something doesn't work 🙏 )

kathleenhang committed:
Update read_csv() with FutureWarning

kathleenhang commented:
Confusion is high. Here are the errors I get from running pre-commit: https://gist.github.com/kathleenhang/edf62b27a5510f96ba4c475f98869c12

Things I've tried:
- Updating from MacOS Monterey -> MacOS Ventura
- Changing Python version in pyproject.toml
- https://bobbyhadz.com/blog/python-preparing-metadata-pyproject-toml-did-not-run-successfully

I'm using Python 3.10.8. I think it might work if I downgrade to Python 3.9. However, that is a struggle too. I get complaints about Linter mypy/flake8 not being installed if I change my Python interpreter in VSCode. 

Also, when I change my Python version in VSCode settings, my Python version is still Python 3.10 within my Docker virtual environment.

Currently I am trying to figure out how to downgrade my Python version within the Docker virtual environment. Is this the way to go or what could be the issue?

kathleenhang committed:
Resolving pre-commit-hook changes

kathleenhang commented:
I ran pre-commit in VSCode which passed all code checks for all modified files. However, it fails the Github code checks. 

I also got this complaint when trying to commit my changes: `[ERROR] Your pre-commit configuration is unstaged.` It wouldn't let me commit without adding the pre-commit configuration file to staging. I wasn't sure how to override the complaint.

MarcoGorelli commented:
Thanks for sticking with this! We're getting close

Having done `pre-commit autoupdate`, CI is now getting extra warnings from the updated versions of the tools you've updated

Was it only `isort` which wasn't working on your Apple? If so, perhaps you could just bump the version of `isort` in `.pre-commit-config.yaml`, and keep the others as they are?

It looks like the `black` formatter isn't passing though - did you stage and commit it after you'd ran `pre-commit`? If not, you can just run
```
pre-commit run --files pandas/tests/io/parser/test_parse_dates.py
```
and then `git add`, `git commit`, `git push`

kathleenhang committed:
Revert pre-commit-config.yaml

kathleenhang commented:
Correct, it was only `isort` which was not working. I was on the latest isort 5.1.2. I'm not sure why the `black` formatter failed, but I reset my staging area and added the files back to the staging area which seems to pass black.

Okay, I've reverted back to the original pre-commit and did as you mentioned. It has passed all checks on my end.

MarcoGorelli reviewed (approved):
Nice, well done @kathleenhang ! Looks good to me pending green

![](https://i.fluffy.cc/MKjDMmwXxZlqRTwnn070gVqZ8Rwh6d3p.gif)

kathleenhang commented:
@MarcoGorelli Thank you!

![plankton-slow-clap](https://user-images.githubusercontent.com/2395780/215495526-36bca8b8-09b9-47b5-93b6-097c1da9c123.gif)

phofl commented:
We have to deploy the isort fix on 1.5.x as well

MarcoGorelli commented:
you're right, thanks - I should've made a separate PR just for that so we could just have backported it

I've opened https://github.com/pandas-dev/pandas/pull/51069

phofl commented:
Thx
